### DISCUSSION TITLE
[API] `createConnection` results in timeout if the length of streams is slightly larger than 4/5.

### DISCUSSION THREAD

**Discussion by ashutosh-015** [airbytehq/airbyte#31557] [Connector Questions]
We're trying to use the Airbyte API on OS Deployment.

- We can create a source successfully.
- We can create a destination successfully.
- While creating the connection using these two, if we have a stream length is 3 or 4 streams, it works.
- When the createConnection API is provided with a larger stream size, it fails and returns timeout.
- We could change the defaultReadTimeOut using the configuration of Airbyte to something like 150s or maybe 1000s, now it just takes 150s to timeout or 1000s to timeout.

This issue has been there from Day 1 of release of Airbyte API to the Open source.

**Comment by corbin-Tprime** [comment]
I also ran into this issue, but I experienced a timeout at `> 10` connections.

**Reply by pgrant87** [reply]
Hi @ashutosh-015 I'm looking into this in conjunction with issue #31472 

Can I ask, what version of Airbyte did you experience this in and what connectors/versions were you using?

Would you mind passing on your request to the "create a connection" endpoint with any sensitive information redacted?

[a comment by an automated account is not shown]
